src/Billing/BillingProcessor/Tasks/GeneratorHCFA_PDF.php

```diff
@@ -0,0 +1,221 @@
+
+    /**
+     * When we run the execute function on each claim, we don't want
+     * to create a new page the first time. The instantiation of the PDF
+     * object "comes with" a canvas to write to, so the first claim, we
+     * don't need to create one. On subsequent claims, we do so we initialize
+     * this to false, and then set to true after the first claim.
+     *
+     * @bool
+     */
+    protected $createNewPage;
+
+    /**
+     * This function is called by the BillingProcessor before the main
+     * claim loop starts.
+     *
+     * Here we set up our PDF canvas and our batch file.
+     *
+     * @param array $context
+     */
+    public function setup(array $context)
+    {
+        $post = $context['post'];
+        $this->pdf = new \Cezpdf('LETTER');
+        $this->pdf->ezSetMargins(trim($post['top_margin']) + 0, 0, trim($post['left_margin']) + 0, 0);
+        $this->pdf->selectFont('Courier');
+
+        // This is to tell our execute method not to create a new page the first claim
+        $this->createNewPage = false;
+
+        // Instantiate mainly for the filename creation, we're not tracking text segments
+        // since we're generating a PDF, which is managed in this object
+        $this->batch = new BillingClaimBatch('.pdf');
+    }
+
+    /**
+     * Do the work to append the given claim to the PDF document we're
+     * working on generating
+     *
+     * @param BillingClaim $claim
+     */
+    protected function updateBatch(BillingClaim $claim)
+    {
+        // Do the actual claim processing
+        $log = '';
+        $hcfa = new Hcfa1500();
+        $lines = $hcfa->genHcfa1500($claim->getPid(), $claim->getEncounter(), $log);
+        $this->appendToLog($log);
+        $alines = explode("\014", $lines); // form feeds may separate pages
+        foreach ($alines as $tmplines) {
+            // The first claim we don't create a new page.
+            if ($this->createNewPage) {
+                $this->pdf->ezNewPage();
+            } else {
+                $this->createNewPage = true;
+            }
+            $this->pdf->ezSetY($this->pdf->ez['pageHeight'] - $this->pdf->ez['topMargin']);
+            $this->pdf->ezText($tmplines, 12, array(
+                'justification' => 'left',
+                'leading' => 12
+            ));
+        }
+    }
+
+    /**
+     *
+     * Do validation, mark claim as billed and tell the billing table
+     * that the claim will be written to the given batch file name
+     *
+     * @param BillingClaim $claim
+     */
+    public function generate(BillingClaim $claim)
+    {
+        // Validate and mark as 'billed'
+        $this->validateAndClear($claim);
+
+        // Finalize the claim
+        if (!BillingUtilities::updateClaim(false, $claim->getPid(), $claim->getEncounter(), -1, -1, 2, 2, $this->batch->getBatFilename())) {
+            $this->printToScreen(xl("Internal error: claim ") . $claim->getId() . xl(" not found!") . "\n");
```

Comment:
is the printToScreen stuff going to the html view?

Reply:
Yes, it gets printed to the modal